src/components/NodeToolbar/Photos/PhotoNotification.js

```diff
@@ -0,0 +1,130 @@
+// @flow
+
+import styled from 'styled-components';
+import * as React from 'react';
+import colors from '../../../lib/colors';
+import { t } from 'c-3po';
+import CheckmarkIcon from '../../icons/actions/CheckmarkIcon';
+import ProblemIcon from '../../icons/actions/ProblemIcon';
+
+type Props = {
+  className: string,
+  notificationType: 'uploadProgress' | 'uploadFailed' | 'reported' | 'waitingForReview';
+  uploadProgress?: number,  // between 0 and 100
+};
+
+type State = {
+};
+
+const defaultState: State = {};
+
+
+const StyledCheckmarkIcon = styled(CheckmarkIcon)`
+  path { 
+    fill: ${props => props.color};
+  }
+`;
+
+const StyledProblemIcon = styled(ProblemIcon)`
+  path { 
+    stroke: ${props => props.color};
+  }
+  circle {
+    fill: ${props => props.color};
+  } 
+`;
+
+class PhotoNotifcation extends React.Component<Props, State> {
+  props: Props;
+  state: State = defaultState;
+
+  componentDidMount() {
+  }
+
+  componentWillReceiveProps(newProps: Props) {
+  }
+
+  render() {
+    const { className, notificationType, uploadProgress } = this.props;
+
+    const uploadProgressMode = notificationType === 'uploadProgress';
+    const uploadFailedMode = notificationType === 'uploadFailed';
+    const reportedMode = notificationType === 'reported';
+    const waitingForReviewMode = notificationType === 'waitingForReview';
+
+    return (
+      <div className={`${className} notification-mode-${notificationType}`}>
+        {uploadProgressMode && <small>
+          <progress max={100} value={uploadProgress || 0} />
+          {t`Upload in Progress...`}</small>
+        }
+        {uploadFailedMode && <small>
+          <StyledProblemIcon color={colors.negativeColor} />
+          {t`Upload failed: server error or file-format not supported`}</small>
+        }
+        {reportedMode && <small>
+          <StyledCheckmarkIcon color={colors.negativeColor} />
+          {t`Thanks for reporting this photo. We will take a look.`}</small>
+        }
+        {waitingForReviewMode && <small>
+          <StyledCheckmarkIcon color={colors.primaryColorBrighter} />
+          {t`Thank you for your work. Your contribution will be visible after a quick check.`}</small>
+        }
+      </div>
```

Comment:
As the modes are exclusive from what I see, something like this would be nicer:

```javascript
const notificationTypesToElements = {
  uploadProgress: <small>…</small>,
  uploadFailed: <small>…</small>,
  reported: <small>…</small>,
  waitingForReview: <small>…</small>,
};

const shownElement = notificationTypesToElements[notificationType];
```

Reply:
Looks better!